internal/internal_workflow.go

```diff
@@ -763,6 +763,23 @@ func (c *channelImpl) Receive(ctx Context, valuePtr interface{}) (more bool) {
 
 }
 
+func (c *channelImpl) ReceiveWithTimeout(ctx Context, timeout time.Duration, valuePtr interface{}) (more bool, timedOut bool) {
```

Comment:
This looks pretty easy to write as a user, and there's no equivalent for this in the Go spec for channels like there is for the other functions here. I might have used timer + selector, but it's basically the same thing.

But if this helper is worth changing the core interface, ok. Alternatively we could just have a package-level function helper for this so the interface is not changed, but it doesn't matter much.

Reply:
I don't have a strong opinion here, this works for me (though would prefer `Empty` to become `Len` if possible).